pkg/messaging/direct_messaging.go

```diff
@@ -194,6 +206,11 @@ func (d *directMessaging) addForwardedHeadersToMetadata(req *invokev1.InvokeMeth
 }
 
 func (d *directMessaging) getAddressFromMessageRequest(appID string) (string, error) {
-	request := nr.ResolveRequest{ID: appID, Namespace: d.namespace, Port: d.grpcPort}
+	namespace, err := d.requestNamespace(appID)
```

Comment:
The concept of namespace is used in K8s in particular so that it depends on the platform. It may make sense to let k8s nameresolver component contrib parse and use appID string while passing default namespace. how do you think ?

Reply:
We had a discussion for this, other platforms have similar concepts.

We even have the namespace concept for self hosted mode today, although we are not enabling it for s2s. But if we wanted to enable it in the future, we should keep this way.

Reply:
Having `namespace` concept is ok. We may need to document it specifically because it could vary by the platform. 

> We even have the namespace concept for self hosted mode today.
Can you elaborate this more ?

Reply:
Developers today can assign namespaces to self-hosted Dapr instances and use it as a [namespace scope for components](https://github.com/dapr/docs/tree/master/howto/components-scopes#example-of-component-namsespacing-in-self-hosted-mode).

It would be very easy to have the mDNS resolver publish the namespace of the instance and enable it for self-hosted s2s invocation too.